lib/portage/util/_async/PipeLogger.py

```diff
@@ -120,8 +131,32 @@ def _output_handler(self, fd):
 								fcntl.F_GETFL) ^ os.O_NONBLOCK)
 
 				if log_file is not None:
-					log_file.write(buf)
-					log_file.flush()
+					write_buf = buf
+					while write_buf:
+						try:
+							# Use os.write, since the log_file.write method
+							# looses data when an EAGAIN occurs.
+							write_buf = write_buf[os.write(log_file.fileno(), write_buf):]
```

Comment:
This broke FEATURES=compress-build-logs, but it can be fixed by offloading compression to a gzip process (https://bugs.gentoo.org/712758), so I'll look into implementing that now.

Reply:
There's a minimal fix in 54cdf7e6cf9ec2030283ab23d659e10562267c6b.